### PR TITLE
VICE 2.4.24: Added recipe and patches

### PR DESCRIPTION
**Pull request by lich**
Commit message:
`The game data goes to /settings/VICE as the files should be edited by user.`

GCI Task: https://codein.withgoogle.com/dashboard/task-instances/5113174395191296/?sp-page=1

### CODE REVIEW

File: games-emulation/vice/vice-2.4.24.recipe [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+GLOBAL_WRITABLE_FILES="
+	settings/VICE directory keep-old
+	"
+
+BUILD()
+{
+	./autogen.sh
+	runConfigure ./configure
+	make
```
Comment (diversys) [inline games-emulation/vice/vice-2.4.24.recipe]: make $jobArgs

File: games-emulation/vice/patches/vice-2.4.24.patchset [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ #endif
+ 
+ #include "archdep.h"
+@@ -122,8 +124,14 @@ static char *boot_path = NULL;
+ const char *archdep_boot_path(void)
+ {
+     if (boot_path == NULL) {
++        #ifdef __HAIKU__
++           char haiku_path[B_PATH_NAME_LENGTH];
++           find_directory(B_SYSTEM_SETTINGS_DIRECTORY, 0, true, haiku_path, sizeof(haiku_path)-strlen("/VICE/"));
```
Comment (diversys) [inline games-emulation/vice/patches/vice-2.4.24.patchset]: Maybe B_USER_NONPACKAGED_DATA_DIRECTORY is a better choice?

Comment (lich) [inline games-emulation/vice/patches/vice-2.4.24.patchset]: Still the files have to be installed and should be edited. The only directories that has that features is /settings/

Comment (diversys) [inline games-emulation/vice/patches/vice-2.4.24.patchset]: NONPACKAGED is editable

Comment (lich) [inline games-emulation/vice/patches/vice-2.4.24.patchset]: But the files from package cant go to nonpackaged.

File: games-emulation/vice/vice-2.4.24.recipe [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	2007-2011 Hannu Nuotio
+	1999-2007 Andreas Dehmel
+	2003-2005 David Hansel
+	2000-2004 Markus Brenner
+	1999-2004 Thomas Bretz
+	1997-2001 Daniel Sladic
+	1996-1999 Ettore Perazzoli
+	1996-1999 Andre Fachat
+	1993-1994, 1997-1999 Teemu Rantanen
+	1993-1996 Jouko Valta "
```
Comment (KapiX) [inline games-emulation/vice/vice-2.4.24.recipe]: Remove the space at the end.

File: games-emulation/vice/vice-2.4.24.recipe [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+SOURCE_URI="http://sourceforge.net/projects/vice-emu/files/development-releases/vice-2.4.24.tar.gz"
+CHECKSUM_SHA256="24e37b4fa268d6034517e41f1802a311af35a5c3a31059c84b6c53e33d5cb59d"
+PATCHES="vice-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	vice$secondaryArchSuffix = $portVersion
+	cmd:c1541g
```
Comment (KapiX) [inline games-emulation/vice/vice-2.4.24.recipe]: Why is there 'g' at the end of each command?

Comment (lich) [inline games-emulation/vice/vice-2.4.24.recipe]: Whoops probably regex error, fixing both things ASAP

**Pull request merged**